spec/Section 7 -- Response.md

```diff
@@ -49,44 +50,47 @@ If an error was raised during the execution that prevented a valid response, the
 
 ### Errors
 
-The `errors` entry in the response is a non-empty list of errors, where each
-error is a map.
+The `errors` entry in the response is a non-empty list of errors raised during
+the _request_, where each error is a map of data described by the error result
+format below.
 
-If no errors were raised during the request, the `errors` entry should not be
+If present, the `errors` entry in the response must contain at least one error.
+If no errors were raised during the request, the `errors` entry must not be
 present in the result.
 
-If the `data` entry in the response is not present, the `errors` entry in the
-response must not be empty. It must contain at least one error. The errors it
-contains should indicate why no data was able to be returned.
+If the `data` entry in the response is not present, the `errors` entry must be
+present. It must contain at least one _request error_ indicating why no data was
+able to be returned.
 
 If the `data` entry in the response is present (including if it is the value
-{null}), the `errors` entry in the response may contain any field errors that
-were raised during execution. If field errors were raised during execution, it
-should contain those errors.
+{null}), the `errors` entry must also be present if a _field error_ was raised
+during execution.
```

Comment:
context:
```
If the `data` entry in the response is present (including if it is the value
```


If we use "if and only if" then it would also enforce that errors must not be present if there were no field errors; I think this adds clarity.


```suggestion
{null}), the `errors` entry must be present if and only if a _field error_ was
raised during execution.
```

We could also change "a" to "one or more":


```suggestion
{null}), the `errors` entry must be present if and only if one or more _field
error_ were raised during execution.
```